File: python/sorting/selection_sort.py
def selection_sort(arr):
    n = len(arr)

    for i in range(n):
        # Assume the current position has the minimum
        min_index = i

        # Find the index of the minimum element
        for j in range(i + 1, n):
            if arr[j] < arr[min_index]:
                min_index = j

        # Swap only if needed (optimization)
        if min_index != i:
            arr[i], arr[min_index] = arr[min_index], arr[i]

    return arr


# Example usage
data = [64, 25, 12, 22, 11]
sorted_data = selection_sort(data)
print("Sorted array:", sorted_data)

File: cpp/sorting/selection_sort.cpp
#include <iostream>
#include <vector>
using namespace std;

void selectionSort(vector<int> &arr){
    int n = arr.size();
    for(int i=0; i<n-1; i++){
        int min_index = i;
        for(int j=i+1; j<n; j++){
            if(arr[j] < arr[min_index])
                min_index = j;
        }
        if(min_index != i) swap(arr[i], arr[min_index]);
    }
}

void printArray(const vector<int> &arr){
    for(int num: arr) cout << num << " ";
    cout << endl;
}

int main(){
    vector<int> arr = {64, 34, 25, 12, 22, 11, 90};
    cout << "Original: ";
    printArray(arr);

    selectionSort(arr);
    cout << "Selection Sorted: ";
    printArray(arr);

    return 0;
}

